test/parallel/test-fs-opendir.js

```diff
@@ -287,3 +288,54 @@ doConcurrentAsyncMixedOps().then(common.mustCall());
   dir.closeSync();
   assert.rejects(dir.close(), dirclosedError).then(common.mustCall());
 }
+
+test('fs.opendir should not double callback if user callback throws', async (t) => {
+  let readCallbackInvokedCount = 0;
+  let userErrorThrown = false;
+
+  const dir = await fs.promises.opendir('.');
+
+  t.after(() => dir.close().catch((err) => console.error('Error closing dir in t.after:', err)));
+
+  await new Promise((resolve, reject) => {
+    const operationTimeout = setTimeout(() => {
+      if (userErrorThrown && readCallbackInvokedCount === 1) {
+        resolve(); // User error handled, no double callback
+      } else if (readCallbackInvokedCount === 0) {
+        reject(new Error('dir.read callback was never invoked.'));
+      } else {
+        reject(new Error('Test timeout reached in an unexpected state.'));
+      }
+    }, 1000);
+
+    dir.read((err, dirent) => {
+      readCallbackInvokedCount++;
+
+      if (err) {
+        // This is an error from dir.read() itself, not the user error we're testing.
+        clearTimeout(operationTimeout);
+        return reject(new Error(`dir.read reported an error: ${err.message}`));
+      }
+
+      if (readCallbackInvokedCount === 1) {
+        userErrorThrown = true;
+        // Simulate the user's code throwing an error.
+        // A robust fs.Dir should catch this internally and not call this callback again.
+        throw new Error('Simulated user error in dir.read callback');
```

Comment:
This is not correct: fs.Dir should not catch and suppress errors from user-provided callback, this error must bubble up.

The test shows that the simulated error disappears on its way, probably due to how `FSReqCallback`'s `oncomplete` is handled on C++ side.

We might need to either investigate the cause and make errors inside `oncomplete` be thrown, or detach the callback call from it (e.g. using `queueMicrotask`).